src/Speller.php

```diff
@@ -146,16 +150,19 @@ public static function spellCurrency($amount, $language, $currency, $requireDeci
 		
 		$text = trim($speller->parseInt($wholeAmount, false, $currency))
 			. ' '
-			. $speller->getCurrencyName('whole', $wholeAmount, $currency);
+			. ($currencyAsCode 
+				? $currency 
+				: $speller->getCurrencyName('whole', $wholeAmount, $currency));
 		
 		if ($requireDecimal || ($decimalAmount > 0))
 		{
-			$text .= $speller->decimalSeparator
-				. ($spellDecimal
-					? trim($speller->parseInt($decimalAmount, true, $currency))
-					: $decimalAmount)
-				. ' '
-				. $speller->getCurrencyName('decimal', $decimalAmount, $currency);
+			$text .= ($currencyAsCode 
+				? ' ' . $decimalAmount . '/100'
```

Comment:
Also, the conditional nesting is getting out of hand now, this should be turned into:
```
if ($decimalAsFraction)
{
    $text .= ' ' . $decimalAmount . '/100';
}
else
{
    $text .= $speller->decimalSeparator;

    if ($spellDecimal)
    {
        $text .= trim($speller->parseInt($decimalAmount, true, $currency);
    }
    else
    {
        $text .= $decimalAmount;
    }
    
    $text .= ' ' . $speller->getCurrencyName('decimal', $decimalAmount, $currency));
}
```